rosie/src/main/java/com/karumi/rosie/domain/usecase/UseCaseHandler.java

```diff
@@ -29,6 +33,10 @@
   private final TaskScheduler taskScheduler;
   private final ErrorHandler errorHandler;
 
+  // This list is used to retain error adapters in a non-weak reference associated with this use
+  // case handler
+  private final List<OnErrorCallback> errorCallbacks = new ArrayList<>();
```

Comment:
The instances of ``OnErrorCallback`` most of the times are implemented using an anonymous function in the view keeping the context of the presenter or the view implementation. Could this be a problem in the future?